go/kbcmf/encrypt.go

```diff
@@ -160,10 +217,16 @@ func (es *encryptStream) init(sender BoxSecretKey, receivers [][]BoxPublicKey) e
 				return err
 			}
 
-			rkc := receiverKeysCiphertext{
-				KID:  kid,
-				Keys: ptec,
+			rkc := receiverKeysCiphertexts{
+				Keys:   ptec,
+				Sender: ske,
 			}
+
+			// Don't specify the receivers if this public key wants to hide
+			if !receiver.HideIdentity() {
+				rkc.KID = receiver.ToKID()
```

Comment:
This makes anonymous receivers distinguishable from visible ones by length. If the sender is the only anonymous receiver with no MAC key, this leaks the sender's number of devices.